libbeat/beat/beat.go

```diff
@@ -141,65 +124,114 @@ func init() {
 	rand.Seed(seed)
 }
 
-// newInstance creates and initializes a new Beat instance.
-func newInstance(name string, version string, bt Beater) *instance {
-	if version == "" {
-		version = defaultBeatVersion
+// Run initializes and runs a Beater implementation. name is the name of the
+// Beat (e.g. packetbeat or metricbeat). version is version number of the Beater
+// implementation. bt is the `Creator` callback for creating a new beater
+// instance.
+func Run(name, version string, bt Creator) error {
+	return handleError(newBeat(name, version).launch(bt))
+}
+
+// newBeat creates a new beat instance
+func newBeat(name, version string) *Beat {
+	return &Beat{
+		Name:    name,
+		Version: version,
+		UUID:    uuid.NewV4(),
+	}
+}
+
+func (b *Beat) launch(bt Creator) error {
+	if err := b.handleFlags(); err != nil {
+		return err
+	}
+
+	svc.BeforeRun()
+	defer svc.Cleanup()
+
+	if err := b.configure(); err != nil {
```

Comment:
We loose some flexibilty here in that no special tasks can be execute by a Beater between configuring the beat and running the beat. This was previously possible. But as the refactoring just worked, I assume this possibility was not needed.